### DISCUSSION TITLE
A good place or approach to get relevant tickers.

### DISCUSSION THREAD

**Discussion by kschmid** [ranaroussi/yfinance#1720] [General]
An issue I deal with is the need to get relevant tickers for major indices. Typical examples are of course: NASDAQ, SP500, but it would also be great to get indices for DAX, MDAX, STOXX, etc. 
Unfortunately, even Yahoo itself only shows some components (and sometimes no components). There are some solutions, which work for individual indices like NASDAQ, but it is a pain to get this from many different sources. 
The best I found so far is: https://github.com/portfolioplus/pytickersymbols
But a lot of this is outdated or provides strange results. So, this is not a good thing, either. 
Does anyone on here, have a good proposal on what to use? (Basically, I am interested in the top few hundred most important US and European stocks.)

**Comment by ms82494** [comment]
If you're willing to pay for it, masterdata.com has constituent lists for a huge number of ETFs/Indexes.

If you're looking for a free alternative, public libraries of some major cities and universities provide remote internet access to Mergent Online, a subsidiary of FTSE/Russell, which maintains downloadable constituent lists for a couple dozen equity indexes.

You can also see the lists on Investing.com. Just type the name of the index into the search bar at the top, hit enter, then select "Constituents" from the page of the index. The paid version of Investing.com (also marketed as finbox.com) **does** permit the download including ticker symbols, but it's not updated as regularly as I would like.

There's also some ambiguity: I've seen constituent lists for the S&P 500 that have well below 500 entries, and others that have over 500. The major source of confusion is whether or not to include the secondary common share classes (both GOOG and GOOGL, or only one). I am not sure that there's a universally accepted way to deal with that.

Personally, I find that the best, most durable way of dealing with this, is to web scrape the sites of index ETFs. Their holding pages get updated frequently, and their layout and web address seems to rarely change. 

It would definitely be nice if someone maintained a repo that collected scraping code for various indexes/ETFs.
